### PR TITLE
Setting up rules and aiPlayer

### PR DESCRIPTION
**Pull request by martiwap**

### CODE REVIEW

File: spec/GameSpec.php [hunk 1]
```diff
@@ -6,8 +6,15 @@
 
 class GameSpec extends ObjectBehavior
 {
-    function it_returns_false_if_no_ships_on_field()
+    function it_returns_score()
     {
-        $this->isSpaceShipOnField()->shouldReturn(false);
+        $this->setScore()->shouldReturn($this);
```
Comment (dblencowe) [inline spec/GameSpec.php]: Should you not check return type here as well?

File: src/AISpaceship.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public function setDamageModifier(int $damageModifier)
+    {
+        $this->damageModifier = $damageModifier;
+
+        return $this;
+    }
+
+    public function setHealth(int $health)
```
Comment (dblencowe) [inline src/AISpaceship.php:29]: You could type hint the returns in here

File: src/Game.php [hunk 3]
```diff
@@ -2,11 +2,31 @@
...
+
+    public function getWinner()
+    {
+        return $this->winner;
+    }
+
+    public function isAnybodyHere(array $shipsOnField) : Boolean
     {
-       if (!empty($shipsOnField)) {
+       if ($shipsOnField) {
```
Comment (dblencowe) [inline src/Game.php]: This could be simplified to `return empty($shipsOnField)`

File: src/GameRulesInterface.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Spaceship;
+
+interface GameRulesInterface
```
Comment (dblencowe) [inline src/GameRulesInterface.php:5]: This interface is correct, but will you need it? Will there be many types of GameRules?

File: spec/GameSpec.php (hunk 1 above)
Comment (martiwap) [reply spec/GameSpec.php]: like? you mean if it is true returns a ship?

File: src/AISpaceship.php (hunk 2 above)
Comment (martiwap) [reply src/AISpaceship.php:29]: : integer
?

File: src/GameRulesInterface.php (hunk 4 above)
Comment (martiwap) [reply src/GameRulesInterface.php:5]: I wasn't sure if I needed it or not, it was more to see if I could have used an interface or not. I could use an abstract class. Or do you suggest just to use a normal class?

File: src/AISpaceship.php (hunk 2 above)
Comment (martiwap) [reply src/AISpaceship.php:29]: If I type hint the return, then the 'return $this' will be an error. In fact, I could type hint : AISpacecship. But I don't know if it is correct..

**Pull request closed without merging**